lib/model/channel.dart

```diff
@@ -203,6 +235,34 @@ class ChannelStoreImpl with ChannelStore {
   @override
   final Map<int, Subscription> subscriptions;
 
+  /// Maps indexed by stream IDs, of the known latest message IDs in each topic.
+  ///
+  /// For example: `_latestMessageIdsByStreamTopic[stream.id][topic] = maxId`
+  ///
+  /// In some cases, the same message IDs, when affected by message moves, can
+  /// be present for mutliple stream-topic keys.
+  final Map<int, Map<TopicName, int>> _latestMessageIdsByStreamTopic = {};
+
+  @override
+  Future<void> fetchTopics(int streamId) async {
+    final result = await _apiGetStreamTopics(connection, streamId: streamId,
+      allowEmptyTopicName: true);
```

Comment:
What if we've already successfully fetched the list for this channel? In that case I think we want to show the existing list (which, after all, this class will have been keeping up to date) rather than make a new fetch.